documents/compute_fields.go

```diff
@@ -143,3 +172,63 @@ func toComputeFieldsAttribute(attr Attribute) (cattr computeAttribute, err error
 
 	return cattr, err
 }
+
+// ExecuteComputeFields executes all the compute fields and updates the document with target attributes
+// each WASM is executed at a max of timeout duration.
+func (cd *CoreDocument) ExecuteComputeFields(timeout time.Duration) error {
+	computeFields := cd.GetComputeFieldsRules()
+
+	ncd := cd
+	for _, computeField := range computeFields {
+		targetAttr, err := executeComputeField(computeField, ncd.Attributes, timeout)
+		if err != nil {
+			return err
+		}
+
+		ncd, err = ncd.AddAttributes(CollaboratorsAccess{}, false, nil, targetAttr)
+		if err != nil {
+			return err
+		}
+	}
+
+	*cd = *ncd
+	return nil
+}
+
+func executeComputeField(rule *coredocumentpb.TransitionRule, attributes map[AttrKey]Attribute, timeout time.Duration) (result Attribute, err error) {
+	var attrs []Attribute
+
+	// filter attributes
+	for _, attr := range rule.ComputeFields {
+		key, err := AttrKeyFromBytes(attr)
+		if err != nil {
+			return result, err
+		}
+
+		fa, ok := attributes[key]
+		if !ok {
+			continue
+		}
+
+		attrs = append(attrs, fa)
+	}
+
+	// execute WASM
+	r := executeWASM(rule.ComputeCode, attrs, timeout)
```

Comment:
What would be the impact down the line of the result being 0..0 value due to an error in the wasm execution? Meaning that since we cannot distinguish between a succcessful run with 0..0 return value and an error. 
That means that we will be pushing this check for validation at the p2p consensus, then right?
What if all the participants run the code and it fails in all of them and then we store 0..0 in the oracle?

Reply:
We want the default value to be zero in case the WASm is failed to be executed or timeout. Oracle fetching the value zero is okay as well since the value above zero is what we expect for NFT feeds